core/src/main/java/org/opensearch/remote/metadata/common/SdkClientUtils.java

```diff
@@ -9,25 +9,210 @@
+import org.opensearch.action.bulk.BulkResponse;
+import org.opensearch.action.delete.DeleteResponse;
+import org.opensearch.action.get.GetResponse;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.update.UpdateResponse;
 import org.opensearch.common.action.ActionFuture;
 import org.opensearch.common.util.concurrent.FutureUtils;
 import org.opensearch.common.util.concurrent.UncategorizedExecutionException;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.remote.metadata.client.BulkDataObjectResponse;
+import org.opensearch.remote.metadata.client.DeleteDataObjectResponse;
+import org.opensearch.remote.metadata.client.GetDataObjectResponse;
+import org.opensearch.remote.metadata.client.PutDataObjectResponse;
+import org.opensearch.remote.metadata.client.SearchDataObjectResponse;
+import org.opensearch.remote.metadata.client.UpdateDataObjectResponse;
 
+import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Locale;
 import java.util.concurrent.CompletionException;
 import java.util.concurrent.ExecutionException;
+import java.util.function.BiConsumer;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import static org.opensearch.core.rest.RestStatus.INTERNAL_SERVER_ERROR;
+
 /**
  * Utility methods for client implementations
  */
 public class SdkClientUtils {
 
     private SdkClientUtils() {}
 
+    /**
+     * Wraps the completion of a PUT operation from the SdkClient into a format compatible with an ActionListener.
+     *
+     * @param listener The ActionListener that will receive the parsed IndexResponse or any errors
+     * @param exceptionTypesToUnwrap optional list of exception types to unwrap. Defaults to {@link OpenSearchStatusException} and {@link CompletionException}.
+     * @return A BiConsumer that can be used directly with CompletionStage's whenComplete method
+     */
+    @SafeVarargs
+    public static BiConsumer<PutDataObjectResponse, Throwable> wrapPutCompletion(
+        ActionListener<IndexResponse> listener,
+        Class<? extends Throwable>... exceptionTypesToUnwrap
+    ) {
+        return (r, throwable) -> {
+            if (throwable == null) {
+                try {
+                    IndexResponse indexResponse = r.parser() == null ? null : IndexResponse.fromXContent(r.parser());
+                    listener.onResponse(indexResponse);
+                } catch (IOException e) {
+                    handleParseFailure(listener, "put");
+                }
+            } else {
+                handleThrowable(listener, throwable, exceptionTypesToUnwrap);
+            }
+        };
+    }
+
+    /**
+     * Wraps the completion of a GET operation from the SdkClient into a format compatible with an ActionListener.
+     *
+     * @param listener The ActionListener that will receive the parsed GetResponse or any errors
```

Comment:
[nit] add `@param exceptionTypesToUnwrap optional list of exception types to unwrap. Defaults to {@link OpenSearchStatusException} and {@link CompletionException}.` to all other methods.

Reply:
Oops.  I'll fix that in my next PR.

Reply:
I thought I did that and I'm surprised the javadoc check passed without it.  Maybe I need to add some more strict checkstyle rules like we have at Flow Framework that catches those.

Also in the category of "approval rules" I've added a `backport/main-before-alpha1` branch that I can push to without PRs.  I'm cherry-picking merged PRs to that branch and just pushing to it and manually running the snapshot action, to keep the non-alpha1 SNAPSHOT up to date until the downstream dependencies are all on alpha1.